src/utils/fetchAmmPrices.js

```diff
@@ -159,25 +161,28 @@ const fetchAmmPrices = async (pools, knownPrices) => {
 };
 
 const fetchChainPools = async (chain, pools) => {
-  const provider = new ethers.providers.JsonRpcProvider(MULTICHAIN_RPC[chain]);
-  const multicall = new ethers.Contract(MULTICALLS[chain], MulticallAbi, provider);
+  const multicallContract = fetchContract(MULTICALLS[chain], BeefyPriceMulticall, chain);
+  const lpInfos = await Promise.all(
```

Comment:
will crash whole api here without try/catch if request failed?
need to catch and also fill .totalSupply, .lp0.balance, .lp1.balance with something so it wont crash later in calc.

same for fetchDmmPrices i guess

Reply:
probably better to catch and handle errors on the caller side instead of each and every method calling the rpc

Reply:
these functions were already crashing when errors happened during rpc calls. They just crashed later when the fields that weren't loaded tried to be accessed... 

I'll look into rewriting everything somehow